### ISSUE TITLE
Run Julia, R and Python code in seperate Malt processes

### ISSUE THREAD

**Issue by felixcremer** [PumasAI/QuartoNotebookRunner.jl#224]
We are trying to run Julia, Python and R code in Quarto without any interaction between the different language cells. 
Unfortunately due to the usage of shared libraries in different versions this does not work, see https://github.com/JuliaPy/PythonCall.jl/issues/397.

Could we run the code from different languages in separate Malt processes and thereby circumventing the interaction between the languages.?

@meggart who had the initial idea.

**Comment by MichaelHatherly** [comment]
So cells of different languages would happen to just be rendered in the same output document, but aside from that they would be evaluated in language-specific processes?

**Comment by felixcremer** [comment]
Yes that would help a lot. We thought we could have one QuartoNotebookWorker process per language and still reuse all of the machinery. Maybe this would have to be set by a flag in the notebook header.

**Comment by MichaelHatherly** [comment]
Seems potentially doable. I'm marking this as up for grabs since we won't be actively working on this ourselves (PumasAI) for the foreseeable future.

**Comment by felixcremer** [comment]
Could you give us some pointers where we best launch these processes? I might  give it a try because this would be really helpful for multi language tutorials.

**Comment by jkrumbiegel** [comment]
You'd have to start making this a vector https://github.com/PumasAI/QuartoNotebookRunner.jl/blob/bc028e0091b6772c6532cb34df4937e9ab0fca22/src/server.jl#L4 and go from there.

But this seems to be a lot of complication for little gain to me, as in principle the interop works, there's just conflicts between the different packages that they could potentially solve. QuartoNotebookRunner is not the only package where these conflicts would manifest so the larger ecosystem could benefit if this was solved somehow at the root instead.

I also think that interpolation from the Julia side to the R/Python side would get more complicated with separate processes. Everything is more complicated with multiple processes 😄

**Comment by MichaelHatherly** [comment]
> I might give it a try because this would be really helpful for multi language tutorials.

There isn't even Python support in a release, it's only a very PoC PR, so it definitely should not be tackled until Python is a well supported feature of this package @felixcremer.

> there's just conflicts between the different packages that they could potentially solve

Agreed, that would be the better solution here rather than adding another feature to this package just to get around problems in the ecosystem.

**Comment by jkrumbiegel** [comment]
I'm going to close this issue as we have decided we do not want to take on the larger complexity of our backend code to handle multiple processes correctly in these cases. In a Slack discussion, the possibility was mentioned to maybe move the responsibility of spinning up a separate process for PythonCall to a shim package or PythonCall itself. I will add a note to that effect to https://github.com/JuliaPy/PythonCall.jl/issues/397 to continue the discussion over there.
